journey-map/public/render.mjs

```diff
@@ -0,0 +1,219 @@
+                Object.values(journeyMap).forEach((journeyDef) => {
+                    if (journeyDef.events?.[entryEvent]?.targetState === state) {
+                        journeyDef.events[entryEvent].targetState = `${def.targetState}_${state}`;
+                    }
+                });
+            });
+        }
+    });
+};
+
+// Should match logic in BasicEvent.java
+const resolveEventTarget = (definition, formData) => {
+    // Look for an override for disabled CRIs
+    const disabledCris = formData.getAll('disabledCri');
+    const disabledResolution = Object.keys(definition.checkIfDisabled || {}).find((k) => disabledCris.includes(k));
+    if (disabledResolution) {
+        return resolveEventTarget(definition.checkIfDisabled[disabledResolution], formData);
+    }
+
+    // Look for an override for feature flags
+    const featureFlags = formData.getAll('featureFlag');
+    const featureFlagResolution = Object.keys(definition.checkFeatureFlag || {}).find((k) => featureFlags.includes(k));
+    if (featureFlagResolution) {
+        return resolveEventTarget(definition.checkFeatureFlag[featureFlagResolution], formData);
+    }
+
+    return definition.targetState;
+};
+
+// Render the transitions into mermaid, while tracking the states traced from the initial states
+// This allows us to skip
+const renderTransitions = (journeyMap, formData) => {
+    const states = [...initialStates];
+    const stateTransitions = [];
+
+    for (let i = 0; i < states.length; i++) {
+        const state = states[i];
+        const definition = journeyMap[state];
+        const events = definition.events || definition.exitEvents || {};
+
+        const eventsByTarget = {};
+        Object.entries(events).forEach(([eventName, def]) => {
+            const target = resolveEventTarget(def, formData);
+
+            if (errorStates.includes(target) && !formData.has('includeErrors')) {
+                return;
+            }
+            if (failureStates.includes(target) && !formData.has('includeFailures')) {
+                return;
+            }
+
+            if (!states.includes(target)) {
+                states.push(target);
+            }
+            eventsByTarget[target] = eventsByTarget[target] || [];
+            eventsByTarget[target].push(eventName);
+        });
+
+        Object.entries(eventsByTarget).forEach(([target, eventNames]) => {
+            stateTransitions.push(`    ${state}-->|${eventNames.join('\\n')}|${target}`);
+        });
+    }
+
+    return { transitionsMermaid: stateTransitions.join('\n'), states };
+};
+
+const renderClickHandler = (state, definition) => {
+    if (!definition.response) {
+        definition.response = {};
+    }
+    // Click handler serializes the definition to Base64-encoded JSON to avoid escaping issues
+    return `    click ${state} call onStateClick(${JSON.stringify(state)}, ${btoa(JSON.stringify(definition.response))})`;
```

Comment:
An unexpanded nested journey doesn't have a response defined on it, which means that when selected in the browser, you just get a `{}`. I'm not entirely sure what would be better, but it looks a little confusing. Maybe just what the nested journey is? Although that's pretty obvious from the state name currently. Maybe just the text "A nested journey"?

Reply:
Yeah that's a good point - it'd be nice to describe a bit more what it's doing at the very least.

One thing we could do is add a 'description' property to nested journeys, which display here - I was imagining that in future each node could actually talk about what the thing is (e.g. what is the 'CIC CRI', and maybe a link to some other documentation).

I will probably leave that for another PR, unless you think it'll be really useful now.

Reply:
No, leave it for another PR I think.